### ISSUE TITLE
[QA] Relative URL breaks client and mail notification

### ISSUE THREAD

**Issue by gabi18** [owncloud/core#38888]
Versions:
ownCloud client 2.8.2

cloud.damken.com 10.7 with patch https://github.com/owncloud/core/pull/38639 applied

Steps to reproduce
1. login to damken.cloud.com as user1
2. user1 shares a file or folder with user2
3. start ownCloud client and login as user2
4. go to 'Activity' 'Server Activity' tab and look for the notifications sent from the server 
5. click on the link

Result:
It's not possible to open the link (URL sent is relative).
Also the 'Decline', 'Accept' and 'Dismiss' buttons don't work

![Share-Notification-Client](https://user-images.githubusercontent.com/75983861/123251149-28b0a380-d4eb-11eb-9da7-cf9bda5aa697.png)

**Comment by gabi18** [comment]
The mail notification is broken as well.

![Share-Mail-Kopano](https://user-images.githubusercontent.com/75983861/123251399-73322000-d4eb-11eb-8e3a-74ba25432687.png)

**Comment by JanAckermann** [comment]
@gabi18 you need the latest notification app (or patch https://github.com/owncloud/notifications/pull/333) to get the correct link in the email (this is where the mails are sent).

@micbar @TheOneRing I don't know what to do about the client issue here, client fix?

**Comment by gabi18** [comment]
From client logfile:
```
06-24 11:50:26:414 [ info sync.httplogger ]:	"36eec34c-ae5c-4d9e-bbd0-448d92c08122: Request: GET https://cloud.damken.com/ocs/v2.php/apps/notifications/api/v1/notifications?format=json Header: { OCS-APIREQUEST: true, Authorization: Bearer [redacted], User-Agent: Mozilla/5.0 (Linux) mirall/2.8.2 (build 4326) (Damken Cloud, opensuse-leap-5.3.18-lp152.78-default ClientArchitecture: x86_64 OsArchitecture: x86_64), Accept: */*, X-Request-ID: 36eec34c-ae5c-4d9e-bbd0-448d92c08122, Cookie: oc_sessionPassphrase=HXI7NqQPzC9sYvtW0Yax%2B8E4yoAHT10ku1Nf1%2B7%2FO%2FfXO71LLd0ziZuCQTESEEjMOGhz8%2BXhYiYbpq2GiJlZxtiwSoHSa8AXSjGvyGWqkagnONEhceSSujVF4hwK74AZ; ocuhsyihz2sp=uq2qejgrdfdmsceneevp6t8tbj, } Data: []"

06-24 11:50:26:664 [ info sync.httplogger ]:	"36eec34c-ae5c-4d9e-bbd0-448d92c08122: Response: GET 200 https://cloud.damken.com/ocs/v2.php/apps/notifications/api/v1/notifications?format=json Header: { Date: Thu, 24 Jun 2021 09:50:26 GMT, Server: Apache/2.4.18 (Ubuntu), Strict-Transport-Security: max-age=15768000, X-Content-Type-Options: nosniff, X-XSS-Protection: 0, X-Robots-Tag: none, X-Frame-Options: SAMEORIGIN, X-Download-Options: noopen, X-Permitted-Cross-Domain-Policies: none, Expires: Thu, 19 Nov 1981 08:52:00 GMT, Cache-Control: no-cache, no-store, must-revalidate, Pragma: no-cache, Content-Security-Policy: default-src 'none';manifest-src 'self';script-src 'self' 'unsafe-eval';style-src 'self' 'unsafe-inline';img-src 'self' data: blob:;font-src 'self';connect-src 'self';media-src 'self', ETag: 60e3994e09aed625e88b62dcb774c3ef, Content-Length: 3368, Keep-Alive: timeout=3, max=200, Connection: Keep-Alive, Content-Type: application/json; charset=utf-8, } Data: [{\"ocs\":{\"meta\":{\"status\":\"ok\",\"statuscode\":200,\"message\":null,\"totalitems\":\"\",\"itemsperpage\":\"\"},\"data\":[{\"notification_id\":1262,\"app\":\"files_sharing\",\"user\":\"gmohr\",\"datetime\":\"2021-06-23T14:59:12+00:00\",\"object_type\":\"local_share\",\"object_id\":\"ocinternal:1934\",\"subject\":\"\\\"J\\u00fcrgen Weigert (jw)\\\" shared \\\"von damken.cloud\\\" with you\",\"message\":\"\\\"J\\u00fcrgen Weigert (jw)\\\" invited you to view \\\"von damken.cloud\\\"\",\"link\":\"\\/index.php\\/f\\/17241526\",\"actions\":[{\"label\":\"Decline\",\"link\":\"\\/ocs\\/v1.php\\/apps\\/files_sharing\\/api\\/v1\\/shares\\/pending\\/1934\",\"type\":\"DELETE\",\"primary\":false},{\"label\":\"Accept\",\"link\":\"\\/ocs\\/v1.php\\/apps\\/files_sharing\\/api\\/v1\\/shares\\/pending\\/1934\",\"type\":\"POST\",\"primary\":true}],\"icon\":\"\\/core\\/img\\/actions\\/shared.svg\"},{\"notification_id\":1261,\"app\":\"files_sharing\",\"user\":\"gmohr\",\"datetime\":\"2021-06-23T14:58:14+00:00\",\"object_type\":\"local_share\",\"object_id\":\"ocinternal:1933\",\"subject\":\"\\\"J\\u00fcrgen Weigert (jw)\\\" shared \\\"von cloud.damken.com\\\" with you\",\"message\":\"\\\"J\\u00fcrgen Weigert (jw)\\\" invited you to view \\\"von cloud.damken.com\\\"\",\"link\":\"\\/index.php\\/f\\/17241524\",\"actions\":[{\"label\":\"Decline\",\"link\":\"\\/ocs\\/v1.php\\/apps\\/files_sharing\\/api\\/v1\\/shares\\/pending\\/1933\",\"type\":\"DELETE\",\"primary\":false},{\"label\":\"Accept\",\"link\":\"\\/ocs\\/v1.php\\/apps\\/files_sharing\\/api\\/v1\\/shares\\/pending\\/1933\",\"type\":\"POST\",\"primary\":true}],\"icon\":\"\\/core\\/img\\/actions\\/shared.svg\"},{\"notification_id\":1260,\"app\":\"files_sharing\",\"user\":\"gmohr\",\"datetime\":\"2021-06-23T14:56:17+00:00\",\"object_type\":\"local_share\",\"object_id\":\"ocinternal:1932\",\"subject\":\"\\\"J\\u00fcrgen Weigert (jw)\\\" shared \\\"fuer Gabi\\\" with you\",\"message\":\"\\\"J\\u00fcrgen Weigert (jw)\\\" invited you to view \\\"fuer Gabi\\\"\",\"link\":\"\\/index.php\\/f\\/17241519\",\"actions\":[{\"label\":\"Decline\",\"link\":\"\\/ocs\\/v1.php\\/apps\\/files_sharing\\/api\\/v1\\/shares\\/pending\\/1932\",\"type\":\"DELETE\",\"primary\":false},{\"label\":\"Accept\",\"link\":\"\\/ocs\\/v1.php\\/apps\\/files_sharing\\/api\\/v1\\/shares\\/pending\\/1932\",\"type\":\"POST\",\"primary\":true}],\"icon\":\"\\/core\\/img\\/actions\\/shared.svg\"},{\"notification_id\":1259,\"app\":\"files_sharing\",\"user\":\"gmohr\",\"datetime\":\"2021-06-23T14:50:38+00:00\",\"object_type\":\"local_share\",\"object_id\":\"ocinternal:1931\",\"subject\":\"\\\"J\\u00fcrgen Weigert (jw)\\\" shared \\\"dreikopftile-3d\\\" with you\",\"message\":\"\\\"J\\u00fcrgen Weigert (jw)\\\" invited you to view \\\"dreikopftile-3d\\\"\",\"link\":\"\\/index.php\\/f\\/16749203\",\"actions\":[{\"label\":\"Decline\",\"link\":\"\\/ocs\\/v1.php\\/apps\\/files_sharing\\/api\\/v1\\/shares\\/pending\\/1931\",\"type\":\"DELETE\",\"primary\":false},{\"label\":\"Accept\",\"link\":\"\\/ocs\\/v1.php\\/apps\\/files_sharing\\/api\\/v1\\/shares\\/pending\\/1931\",\"type\":\"POST\",\"primary\":true}],\"icon\":\"\\/core\\/img\\/actions\\/shared.svg\"},{\"notification_id\":1213,\"app\":\"files_sharing\",\"user\":\"gmohr\",\"datetime\":\"2021-05-04T22:25:45+00:00\",\"object_type\":\"local_share\",\"object_id\":\"ocinternal:1867\",\"subject\":\"\\\"J\\u00fcrgen Weigert (jw)\\\" shared \\\"tmp\\\" with you\",\"message\":\"\\\"J\\u00fcrgen Weigert (jw)\\\" invited you to view \\\"tmp\\\"\",\"link\":\"\\/index.php\\/f\\/16646906\",\"actions\":[{\"label\":\"Decline\",\"link\":\"\\/ocs\\/v1.php\\/apps\\/files_sharing\\/api\\/v1\\/shares\\/pending\\/1867\",\"type\":\"DELETE\",\"primary\":false},{\"label\":\"Accept\",\"link\":\"\\/ocs\\/v1.php\\/apps\\/files_sharing\\/api\\/v1\\/shares\\/pending\\/1867\",\"type\":\"POST\",\"primary\":true}],\"icon\":\"\\/core\\/img\\/actions\\/shared.svg\"}]}}]
```

**Comment by jnweiger** [comment]
FYI: There is also a new market app release containing https://github.com/owncloud/market/pull/650

**Comment by JanAckermann** [comment]
Please keep in mind, these changes are dependent!
In order to get rid of errors in any 'Use relative url instead of absoulute url' ticket (in WebUI and Emails), you need the Notifications 0.5.3 app.

**Comment by micbar** [comment]
@jnweiger @JanAckermann @TheOneRing We need to know very soon because 10.8.0 is already in Code freeze.

**Comment by JanAckermann** [comment]
@gabi18 can you confirm that the emails work with notifications app 0.5.3, we tested it and it works

**Comment by JammingBen** [comment]
We will do another update of the notification app to fix the behavior with the client.

**Comment by gabi18** [comment]
> @gabi18 can you confirm that the emails work with notifications app 0.5.3, we tested it and it works

I will retest with with notifications app 0.5.3.

**Comment by gabi18** [comment]
> > @gabi18 can you confirm that the emails work with notifications app 0.5.3, we tested it and it works
> 
> I will retest with with notifications app 0.5.3.

Confirmed that email notification works (has absolute link) with 0.5.3.

**Comment by JanAckermann** [comment]
@gabi18 thank you, with 0.5.4 (still in release phase) client should work again as well, I will inform you :)

**Comment by jnweiger** [comment]
A confusing side note:
Just saw an email notification with a localhost URL:
```
Hello mailuser,

You are receiving this email because the following things happened at http://localhost/

* admin announced Share Notification - Today at 2:12:48 PM
* admin shared Sharefolder with you - Today at 3:51:07 PM
* You created Sharefolder/crypt-o.mp3 - Today at 3:53:37 PM
* admin removed the share for Sharefolder - Today at 3:56:07 PM
* admin shared Photos with you - Today at 3:56:19 PM

--
ownCloud - A safe home for all your data
https://owncloud.org
```
I assume, this is due to not having set overwrite.cli.url set correctly, but seem related :-)

**Comment by JanAckermann** [comment]
I saw this before with notifications 0.5.2, @jnweiger  can you confirm this is 0.5.3 regression?

**Comment by jnweiger** [comment]
```
root@jw-oc-10-8-0beta1-20210625-bg2:/var/www/owncloud/config# occ app:list notifications
Enabled:
  - notifications:
    - Version: 0.5.3
    - Path: /var/www/owncloud/apps/notifications
```

**Comment by gabi18** [comment]
On same server:
![Notification_Mailhog](https://user-images.githubusercontent.com/75983861/123660198-bcf47080-d833-11eb-9a1a-12736c2a7103.png)

**Comment by gabi18** [comment]
Just saw another mail with 'localhost' URL for notification created from announcementcenter:

![annoucement_mail](https://user-images.githubusercontent.com/75983861/123660986-6e93a180-d834-11eb-8766-deb417532897.png)

**Comment by JanAckermann** [comment]
@jnweiger please check if this happens with 0.5.2 as well, I guess these are handled via the activity app and does not have the relative links atm

**Comment by JanAckermann** [comment]
@gabi18 good or bad?

**Comment by JanAckermann** [comment]
@gabi18 which version of announcement enter ist this ?

**Comment by gabi18** [comment]
Sorry for confusion, confirmed that after @jnweiger changed the overwrite.cli.url the link is correct.

**Comment by gabi18** [comment]
![correct_URL](https://user-images.githubusercontent.com/75983861/123662238-8d466800-d835-11eb-98a2-2d32d5188e94.png)

**Comment by JanAckermann** [comment]
Okay, but I still wondering why in some cases the URL is correct and in some cases not, I guess not all of these mails are set with a relative url

**Comment by jnweiger** [comment]
I don't think anything was wrong with these localhost urls. I had forgotten to correctly configure the overwrite.cli.url --- 
But giving that a second thought: Is the concept of an overwrite.cli.url a good thing at all? It cannot really work with a multihomed server, can it? (I don't think that is a regression -- but cannot test with notifications 0.5.2 now; the setup is horribly tedious....)

**Comment by JanAckermann** [comment]
TBH I don't have any idea, but I guess these emails are sent via the activity app, and those links are not relative yet, need to investigate

**Comment by micbar** [comment]
@jnweiger @JanAckermann cli url is always needed.
Activity mails are sent in the background

Background jobs are started via crontab -> cli
CLI has no awareness of url - no apache involved

So all CLI jobs require a `overwrite.cli.url`

I consider an owncloud without it as "nonfunctional"

**Comment by JanAckermann** [comment]
@gabi18 the client issue has been fixed with notifications 0.5.4 👍

**Comment by gabi18** [comment]
Thank you!
Confirmed - tested on https://oc-10-8-0beta1-20210625.jw-qa.owncloud.works/owncloud with notifications 0.5.4 installed -
that client receives notifications containing an absolute path.
